lemur/certificates/verify.py

```diff
@@ -55,21 +57,33 @@ def ocsp_verify(cert, cert_path, issuer_chain_path):
         stderr=subprocess.PIPE,
     )
 
-    message, err = p2.communicate()
+    try:
+        message, err = p2.communicate(timeout=15)
+    except TimeoutExpired:
+        p2.kill()
+        p2.communicate()
```

Comment:
Is this a retry without any timeout?

Reply:
it does indeed look like a try, though my understanding from the [docs](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate) was that after kill, the communicate would finish communication.

> The child process is not killed if the timeout expires, so in order to cleanup properly a well-behaved application should kill the child process and finish communication


This might also be a MS artifact, from online [readings](https://github.com/python/cpython/commit/5fe153c21de020f37712910b0dcce93341bba272)
```
            if _mswindows:
                # Windows accumulates the output in a single blocking
                # read() call run on child threads, with the timeout
                # being done in a join() on those threads.  communicate()
                # _after_ kill() is required to collect that and add it
                # to the exception.
                exc.stdout, exc.stderr = process.communicate()
```


will remove it for now. 

hoping we can just rewrite this in Python, instead of using subprocesses.